packages/medusa/src/api/middlewares/with-default-sales-channel.ts

```diff
@@ -0,0 +1,28 @@
+import { NextFunction, Request, Response } from "express"
+import { SalesChannelService } from "../../services"
+
+/**
+ * Middleware that includes the default sales channel on the request, if no sales channels present
+ * @param asArray Whether to include the default sales channel id as an array or just a string
+ */
+export function withDefaultSalesChannel(
```

Comment:
suggestion: Would it make sense to have that as an util in that file and export `withDefaultSalesChannel` and `withDefaultSalesChannels` and use the right one in the API handlers and it would pass true as the argument if you use withDefaultSalesChannels. wdyt?

Reply:
I thought about that approach as well, but I don't necessarily know that it makes the usage clearer. Imo `withDefaultSalesChannels` would be slightly odd, as there's only one default.

Reply:
Yeah let’s not do that ahah